In [27]:
# Imports
import pandas as pd
import numpy as np
from sklearn import datasets
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split


# Part 1: Load the dataset


In [28]:
# Load the dataset (load remotely, not locally)
# Output the first 15 rows of the data
d = datasets.load_iris()
data = pd.DataFrame(d.data, columns=d.feature_names)
data.head(15)


sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                 5.1               3.5                1.4               0.2
1                 4.9               3.0                1.4               0.2
2                 4.7               3.2                1.3               0.2
3                 4.6               3.1                1.5               0.2
4                 5.0               3.6                1.4               0.2
5                 5.4               3.9                1.7               0.4
6                 4.6               3.4                1.4               0.3
7                 5.0               3.4                1.5               0.2
8                 4.4               2.9                1.4               0.2
9                 4.9               3.1                1.5               0.1
10                5.4               3.7                1.5               0.2
11                4.8               3.4                1.6               0.2
12                4.8               3.0                1.4               0.1
13                4.3               3.0                1.1               0.1
14                5.8               4.0                1.2               0.2

In [29]:
# Display a summary of the table information (number of datapoints, etc.)
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
dtypes: float64(4)
memory usage: 4.8 KB


### About the dataset
Explain what the data is in your own words. What are your features and labels? What is the mapping of your labels to the actual classes? 
* The Iris data set is composed of 150 data samples across 3 different iris variations. Each iris variant have 50 samples that records many features of each individual flower. The labels include Setosa, Veriscolor, and Virginica. As for the features: We have the measurements of the sepal length, sepal width, petal length and petal width.


# Part 2: Split the dataset into train and test


In [32]:
# Take the dataset and split it into our features (X) and label (y)
X = d.data
y = d.target
# Use sklearn to split the features and labels into a training/test set. (90% train, 10% test)
X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.9, test_size=0.1)
# 135 went into train set, 15 
print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)

(135, 4) (15, 4) (135,) (15,)


# Part 3: Logistic Regression

In [ ]:
# i. Use sklearn to train a LogisticRegression model on the training set

# ii. For a sample datapoint, predict the probabilities for each possible class

# iii. Report on the score for Logistic regression model, what does the score measure?

# iv. Extract the coefficents and intercepts for the boundary line(s)

# Part 4: Support Vector Machine


In [1]:
# i. Use sklearn to train a Support Vector Classifier on the training set

# ii. For a sample datapoint, predict the probabilities for each possible class

# iii. Report on the score for the SVM, what does the score measure?

# Part 5: Neural Network


In [ ]:
# i. Use sklearn to train a Neural Network (MLP Classifier) on the training set

# ii. For a sample datapoint, predict the probabilities for each possible class

# iii. Report on the score for the Neural Network, what does the score measure?

# iv: Experiment with different options for the neural network, report on your best configuration (the highest score I was able to achieve was 0.8666)

# Part 6: K-Nearest Neighbors


In [2]:
# i. Use sklearn to 'train' a k-Neighbors Classifier
# Note: KNN is a nonparametric model and technically doesn't require training
# fit will essentially load the data into the model see link below for more information
# https://stats.stackexchange.com/questions/349842/why-do-we-need-to-fit-a-k-nearest-neighbors-classifier

# ii. For a sample datapoint, predict the probabilities for each possible class

# iii. Report on the score for kNN, what does the score measure?

# Part 7: Conclusions and takeaways
In your own words describe the results of the notebook. Which model(s) performed the best on the dataset? Why do you think that is? Did anything surprise you about the exercise?